dora/active_sampling/lin_sampler.py

```diff
@@ -0,0 +1,391 @@
+""" Bayesian Linear Sampler Module. """
+import logging
+
+import numpy as np
+import scipy
+import scipy.stats as stats
+import scipy.linalg as linalg
+
+from dora.active_sampling.base_sampler import Sampler, random_sample
+
+log = logging.getLogger(__name__)
+
+class BayesianLinear(Sampler):
+    """BayesianLinear Class.
+    Attributes
+    ----------
+    basisdef: str
+            basis function definition.
+    basisparams: list [mu, s]
+                basis function parameters.
+    feature: int
+            number of features.
+    n_min: int
+        number of training samples required before sampler is trained.
+    acq_name: str
+            acquisition function definition.
+    explore_priority: float, optional
+                    the priority of exploration against exploitation.
+    hyperparams: list [alpha, beta]
+                hyperparameters of the Bayesian Linear model.
+    y_mean: float
+            mean of the training target outputs.
+    regressors: list
+                list of regressor objects.
+    n_tasks: int
+            number of tasks.
+    """
+
+    name = 'BayesianLinear'
+
+    def __init__(self, lower, upper, basisdef=None, basisparams=None, feature=5, n_train=50,
+                 acq_name="var_sum", explore_priority=1.0, seed=None):
+
+        Sampler.__init__(self, lower, upper)
+        self.basisdef = basisdef
+        self.basisparams = basisparams
+        self.feature =feature
+        self.n_min = n_train
+        self.acq_name = acq_name
+        self.explore_priority = explore_priority
+        self.hyperparams = None
+        self.regressors = None
+        self.y_mean = None
+        self.n_tasks = None
+        if seed:
+            np.random.seed(seed)
+
+
+    def update_y_mean(self):
+        """update the mean of the target outputs."""
+        if not self.y:
+            return
+        self.y_mean = self.y().mean(axis=0) if len(self.y) else None
+        if self.n_tasks is None:
+            self.n_tasks = self.y_mean.shape[0]
+        else:
+            assert self.n_tasks == self.y_mean.shape[0]
+
+
+    def FeatureGen(self, x):
+        """Compute the feature matrix.
+
+        Parameters
+        ----------
+        x: ArrayBuffer
```

Comment:
For arrays, it's good practice to include the expected dimensions (eg. N x D and state what N and D are in the line below)